TypeSystem/flow-analysis/reachability_A22_t05.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion - Method invocation: If `N` is an expression of the form
+/// `E1.m1(E2)`, then:
+/// - Let `before(E1) = before(N)`
+/// - Let `before(E2) = after(E1)`
+/// - Let `T` be the static return type of the invocation
+/// - If `T <: Never` then:
+///   - Let `after(N) = unreachable(after(E2))`.
+/// - Otherwise:
+///   - Let `after(N) = after(E2)`.
+///
+/// @description Checks that if the static type of the expression of the form
+/// `E1.m1(E2)` static type of `E1`is not `Never` then `after(N) = after(E2)`.
```

Comment:
Again, that's a big claim. There could be all kinds of detailed differences between `after(N)` and `after(E2)` that we'd have to write many tests in order to cover. (For example, some innocent bystander of a local variable could suddenly have been promoted, because of a bug, so `after(N) != after(E2)` with respect to that particular variable; and perhaps this only happens when the name of the variable is `xyzzy31` ;-). Perhaps:
```suggestion
/// `E1.m1(E2)` static type of `E1`is not `Never` then `after(N) = after(E2)`,
/// which is tested by detecting that `i = 42` is considered to be guaranteed
/// to have been executed when `i;` is executed.
```
It is probably possible to cut that down to fewer words and still inform the reader about the situation, but the point is just that we can't claim to have shown that `after(N) == after(E2)`, and we don't even want to get anywhere near to being able to claim that.